deployment/promotion/promote.py

```diff
@@ -9,396 +9,157 @@
 - VERTA_SOURCE_DEV_KEY: The dev key associated to the email address on the source Verta instance
 - VERTA_SOURCE_WORKSPACE: The workspace associated with the build on the source Verta instance
+- VERTA_SOURCE_ORGANIZATION: The organization associated with the build on the source Verta instance
+- VERTA_SOURCE_S3_BUCKET: The S3 bucket where the artifacts are stored
 - VERTA_DEST_HOST: The destination Verta instance to promote to
 - VERTA_DEST_EMAIL: The email address for authentication to the destination Verta instance
 - VERTA_DEST_DEV_KEY: The dev key associated to the email address on the destination Verta instance
-- VERTA_DEST_WORKSPACE: The name of the workspace associated with the build on the destination Verta instance
-- VERTA_DEST_REGISTERED_MODEL: [optional] The name of the registered model to promote to. If missing, we'll create a new registered model
-
-Optional environment variables to configure curl usage:
-VERTA_CURL_OPTS: Options to pass to curl. Defaults to '-O'
+- VERTA_DEST_WORKSPACE: The workspace associated with the build on the destination Verta instance
+- VERTA_DEST_ORGANIZATION: The organization associated with the build on the destination Verta instance
+- VERTA_DEST_S3_BUCKET: The S3 bucket where the artifacts will be stored
+- VERTA_DEST_REGISTERED_MODEL_ID: [optional] The ID of the registered model to promote to. If missing, we'll create a new registered model
 
 With these values set, run the script. The script will not attempt to delete any data and will fail if the registered
 model (if an existing one has not been provided) or version already exists in the destination.
 """
 
 from locale import atoi
+from dataclasses import dataclass, field
+from typing import Optional
 
 import requests
 import os
 import datetime
 
-env_vars = ['VERTA_SOURCE_MODEL_VERSION_ID', 'VERTA_SOURCE_HOST', 'VERTA_SOURCE_EMAIL', 'VERTA_SOURCE_DEV_KEY',
-            'VERTA_SOURCE_WORKSPACE', 'VERTA_DEST_HOST', 'VERTA_DEST_EMAIL', 'VERTA_DEST_DEV_KEY',
-            'VERTA_DEST_WORKSPACE']
-opt_env_vars = ['VERTA_DEST_REGISTERED_MODEL']
-params = {}
-
-
-for param_name in env_vars:
-    param = os.environ.get(param_name)
-    if not param:
-        raise KeyError("Missing environment variable %s", param_name)
-    params[param_name] = param
-
-for param_name in opt_env_vars:
-    param = os.environ.get(param_name)
-    params[param_name] = param
-
-curl_opts = os.environ.get('VERTA_CURL_OPTS')
-if curl_opts:
-    params['VERTA_CURL_OPTS'] = curl_opts
+from verta import Client
+
+@dataclass
+class Config:
+    host: str
+    email: str
+    devkey: str
+    organization: Optional[str] = field(default=None)
+    workspace: Optional[str] = field(default=None)
+
+source_config = Config(
+    host=os.environ.get('VERTA_SOURCE_HOST'),
+    email=os.environ.get('VERTA_SOURCE_EMAIL'),
+    devkey=os.environ.get('VERTA_SOURCE_DEV_KEY'),
+    organization=os.environ.get('VERTA_SOURCE_ORGANIZATION'),
```

Comment:
I believe in our customers' case there will only be one organization per environment. Might be worth falling back to their default.

Reply:
Yup, did a quick check and we don't expose the org on the UI if there's only one. You'll definitely need to fix this.

Reply:
But if this is called by a system admin, they _have_ to set it, right? Otherwise we won't infer the organization, I believe. What's the recommendation here?

Reply:
Not if there's only one org, which is true for this customer! Since this script is particular to a particular customer with only one org per env (as opposed to SAAS prod), we'll infer the org.